MC/config/PWGHF/external/generator/generator_pythia8_gaptriggered_hf.C

```diff
@@ -154,16 +174,32 @@ FairGenerator *GeneratorPythia8GapTriggeredCharm(int inputTriggerRatio, float yQ
 }
 
 // Beauty-enriched
-FairGenerator *GeneratorPythia8GapTriggeredBeauty(int inputTriggerRatio, float yMin=-1.5, float yMax=1.5, float yHadronMin=-1.5, float yHadronMax=1.5, int pdgCodeCharmHadron=0, bool doNoQuarkTrigger=false) {
+FairGenerator *GeneratorPythia8GapTriggeredBeauty(int inputTriggerRatio, float yQuarkMin=-1.5, float yQuarkMax=1.5, float yHadronMin=-1.5, float yHadronMax=1.5, int pdgCodeCharmHadron=0, bool doNoQuarkTrigger=false) {
   auto myGen = new GeneratorPythia8GapTriggeredHF(inputTriggerRatio, 5);
   auto seed = (gRandom->TRandom::GetSeed() % 900000000);
   myGen->readString("Random:setSeed on");
   myGen->readString("Random:seed " + std::to_string(seed));
-  myGen->setQuarkRapidity(yMin, yMax);
+  myGen->setQuarkRapidity(yQuarkMin, yQuarkMax);
+  if(pdgCodeCharmHadron != 0) {
+    myGen->setQuarkTrigger(doNoQuarkTrigger);
+    myGen->addTriggerOnHadron(pdgCodeCharmHadron);
+    myGen->setHadronRapidity(yHadronMin, yHadronMax);
+  }
+  return myGen;
+}
+
+// Charm and beauty enriched (with same ratio)
+FairGenerator *GeneratorPythia8GapTriggeredCharmAndBeauty(int inputTriggerRatio, float yQuarkMin=-1.5, float yQuarkMax=1.5, float yHadronMin=-1.5, float yHadronMax=1.5, int pdgCodeCharmHadron=0, bool doNoQuarkTrigger=false) {
+  auto myGen = new GeneratorPythia8GapTriggeredHF(inputTriggerRatio, 4);
```

Comment:
I don't particularly like the fact that we set the quark pdg to the charm only if we alternate: could we maybe use another code here, or maybe pass a vector of pdg codes (size 1 if no alternated, size 2 if alternated).

Reply:
Hi @fgrosa, I didn't get the point here. The new generator I added has no way for the user to change/mess with the alternate injection since it is not exposed in the arguments. The implementation is bare bone but also simple and effective.

Can you clarify or do you have some code suggestions to propose?

Reply:
The code is indeed correct, I was commenting about the clarity of it: when you construct the generator you set the quark pdg code to 4, so it seems that only charm is set. Keep in mind that a user could directly use the `GeneratorPythia8GapTriggeredHF` and not using the predefined functions that return the generator configured. Hence, I was suggesting either to replace the `mPdgCode` initialisation to a different value (e.g. could be `9=4+5`) to be more explicit, or with a vector of integers (initialised to `{4}` or `{5}` in case of a single flavour and `{4, 5}` in case of charm + beauty. This only requires then a small change (i.e. you have to define an independent variable that is used for the actual quark pdg event by event).

Reply:
Ok, I implemented your suggestion

Reply:
Thanks @fcatalan92!